.github/workflows/sync-from-upstream.yml

```diff
@@ -0,0 +1,73 @@
+name: sync-panther-analysis-from-upstream
+on:
+  schedule:
+    # 15:00Z, every Monday
+    - cron: "00 15 * * 1"
+  workflow_dispatch: # or on button click  
+
+jobs:
+  check_upstream:
+    ## Change the false in this if statement to true in order to run this workflow
+    if: | 
+      github.repository != 'panther-labs/panther-analysis' &&
+      false
+    runs-on: ubuntu-latest
+    steps:
+      - name: echo stuff
+        run: |
+          echo "${{ github }}"
+          echo and also
+          echo "${{ context }}"
+      - uses: actions/github-script@v6
+        id: set_upstream
+        name: Check Upstream
+        with:
+          script: |
+            const upstreamLatest = await github.rest.repos.getLatestRelease({
+              owner: 'panther-labs',
+              repo: 'panther-analysis'
+            })
+            console.log("::set-output name=latest-release::" +  upstreamLatest.data.tag_name)
+      ## CREATE A BRANCH
+      - uses: peterjgrainger/action-create-branch@v2.2.0
+        id: create_a_branch
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        with:
+          branch: 'sync_upstream_${{steps.set_upstream.outputs.latest-release}}'
```

Comment:
How does this behave if there is no new release and it tries to create a branch of the same name as the previous?

Reply:
It will do through the branch creation mechanisms, pull upstream, and the pull request step will try and create the PR. This step will report "no changes in branch" and believe it or not it exits successfully ( but no PR is created ).  See line 35 in the `Create Pull Request` step  [in my personal fork](https://github.com/edyesed/panther-analysis/actions/runs/3123852485/jobs/5066774990#step:6:41) from the scheduled run from this morning to see a  real world example.